### DISCUSSION TITLE
Define unions in V to match the C equivalent

### DISCUSSION THREAD

**Discussion by prime31** [vlang/v#11632] [Ideas]
**V version:** V 0.1.23 f7c1b78
**OS:** macos

**What did you do?**
Attempted to define a V struct to match the C struct below:

```c
typedef union hmm_vec2
{
    struct
    {
        float X, Y;
    };

    struct
    {
        float U, V;
    };

    struct
    {
        float Left, Right;
    };

    struct
    {
        float Width, Height;
    };

    float Elements[2];
} hmm_vec2;
```

**What did you expect to see?**
Some way to be able to define the C struct in V so that we can access the C code. The following does not work:

```
struct C.hmm_vec3 { 
	struct {
        X f32
		Y f32
    }

    struct {
        U f32
		V f32
    }
}
```
 
**What did you see instead?**

```
next token = `{`
bug.v:14:5: syntax error: unexpected `struct`, expecting `name` 12| struct C.hmm_vec3 { 13| struct 14| { ^ 15| X f32 16| Y f32
```

**Comment by getfile** [comment]
yes , i met the same question.
vlang now in 0.3, can some one to fit it?

**Comment by km19809** [comment]
Should this be filed as an issue, not a discussion?
